diff --git a/lua-nginx-module/config b/lua-nginx-module/config
index 044deb9..ea35f23 100644
--- a/lua-nginx-module/config
+++ b/lua-nginx-module/config
@@ -281,6 +281,30 @@ END
             fi
             . auto/feature
         fi
+
+        if [ $ngx_found = no ]; then
+            # Ubuntu 18.04 with LuaJIT-2.1, try with -ldl
+            ngx_feature="LuaJIT library in /usr/"
+            ngx_feature_path="/usr/include/luajit-2.1"
+            if [ $NGX_RPATH = YES ]; then
+                ngx_feature_libs="-R/usr/lib -L/usr/lib -lm -lluajit-5.1 -ldl"
+            else
+                ngx_feature_libs="-L/usr/lib -lm -lluajit-5.1 -ldl"
+            fi
+            . auto/feature
+        fi
+
+        if [ $ngx_found = no ]; then
+            # Ubuntu 18.04 with LuaJIT 2.1
+            ngx_feature="LuaJIT library in /usr/"
+            ngx_feature_path="/usr/include/luajit-2.1"
+            if [ $NGX_RPATH = YES ]; then
+                ngx_feature_libs="-R/usr/lib -L/usr/lib -lm -lluajit-5.1"
+            else
+                ngx_feature_libs="-L/usr/lib -lm -lluajit-5.1"
+            fi
+            . auto/feature
+        fi
     fi
 fi
 
