From: Sergey Budnevitch <sb@nginx.com>
Date: Wed, 25 Jul 2018 12:43:19 +0900
Subject: lua-nginx-module-luajit-2.1

---
 lua-nginx-module/config | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

--- a/lua-nginx-module/config
+++ b/lua-nginx-module/config
@@ -1,3 +1,4 @@
+set -x
 ngx_lua_opt_I=
 ngx_lua_opt_L=
 
@@ -132,6 +133,30 @@
         if [ $NGX_RPATH = YES ]; then
             ngx_feature_libs="-R/usr/lib -L/usr/lib -lm -lluajit-5.1"
         else
+            ngx_feature_libs="-L/usr/lib -lm -lluajit-5.1"
+        fi
+        . auto/feature
+    fi
+
+    if [ $ngx_found = no ]; then
+        # Ubuntu 18.04 with LuaJIT-2.1, try with -ldl
+        ngx_feature="LuaJIT library in /usr/"
+        ngx_feature_path="/usr/include/luajit-2.1"
+        if [ $NGX_RPATH = YES ]; then
+            ngx_feature_libs="-R/usr/lib -L/usr/lib -lm -lluajit-5.1 -ldl"
+        else
+            ngx_feature_libs="-L/usr/lib -lm -lluajit-5.1 -ldl"
+        fi
+        . auto/feature
+    fi
+
+    if [ $ngx_found = no ]; then
+        # Ubuntu 18.04 with LuaJIT 2.1
+        ngx_feature="LuaJIT library in /usr/"
+        ngx_feature_path="/usr/include/luajit-2.1"
+        if [ $NGX_RPATH = YES ]; then
+            ngx_feature_libs="-R/usr/lib -L/usr/lib -lm -lluajit-5.1"
+        else
             ngx_feature_libs="-L/usr/lib -lm -lluajit-5.1"
         fi
         . auto/feature
