diff --git a/lua-resty-lmdb/config b/lua-resty-lmdb/config
index 126c78c..a5a2082 100644
--- a/lua-resty-lmdb/config
+++ b/lua-resty-lmdb/config
@@ -3,6 +3,11 @@ ngx_module_name=ngx_lua_resty_lmdb_module
 ngx_module_srcs="$ngx_addon_dir/src/ngx_lua_resty_lmdb_module.c $ngx_addon_dir/src/ngx_lua_resty_lmdb_transaction.c"
 ngx_module_incs="$ngx_addon_dir/lmdb/libraries/liblmdb $ngx_addon_dir/src"
 
+if [ $ngx_module_link = DYNAMIC ] ; then
+    ngx_module_deps="$ngx_addon_dir/lmdb/libraries/liblmdb/liblmdb-lua-resty.so"
+    ngx_module_libs="-L$ngx_addon_dir/lmdb/libraries/liblmdb -llmdb-lua-resty"
+fi
+
 . auto/module
 
 LINK_DEPS="$LINK_DEPS $ngx_addon_dir/lmdb/libraries/liblmdb/liblmdb.a"
diff --git a/lua-resty-lmdb/config.make b/lua-resty-lmdb/config.make
index 14d8cc2..fa83e1a 100644
--- a/lua-resty-lmdb/config.make
+++ b/lua-resty-lmdb/config.make
@@ -2,8 +2,14 @@
 cat <<EOF >>$NGX_MAKEFILE
 
 $ngx_addon_dir/lmdb/libraries/liblmdb/liblmdb.a:
-	echo "Building liblmdb"; \\
+	echo "Building liblmdb static library"; \\
 	\$(MAKE) -C $ngx_addon_dir/lmdb/libraries/liblmdb; \\
-	echo "Finished building liblmdb"
+	echo "Finished building liblmdb static library"
 
-EOF
\ No newline at end of file
+$ngx_addon_dir/lmdb/libraries/liblmdb/liblmdb-lua-resty.so:
+	echo "Building liblmdb shared library"; \\
+	\$(MAKE) XCFLAGS="-fPIC" -C $ngx_addon_dir/lmdb/libraries/liblmdb; \\
+	mv $ngx_addon_dir/lmdb/libraries/liblmdb/liblmdb.so $ngx_addon_dir/lmdb/libraries/liblmdb/liblmdb-lua-resty.so; \\
+	echo "Finished building liblmdb shared library"
+
+EOF
